src/report_workflow/report_args.py

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import argparse
+import logging
+
+from test_workflow.test_kwargs import TestKwargs
+
+
+class ReportArgs:
+    test_run_id: str
+    keep: bool
+    test_manifest_path: str
+    artifact_paths: dict
+    test_type: str
+    logging_level: int
+
+    def __init__(self) -> None:
+        parser = argparse.ArgumentParser(description="Report for test on OpenSearch Bundle")
+        parser.add_argument("test_manifest_path", type=str, help="Specify a test manifest path.")
+        parser.add_argument("-p", "--artifact-paths", nargs='*', action=TestKwargs, default={},
+                            help="Specify aritfacts paths for OpenSearch and OpenSearch Dashboards.")
+        # e.g. --base-path https://ci.opensearch.org/ci/dbc/integ-test/2.7.0/7771/linux/x64/tar/test-results/1234<test-run-id>/integ-test use more to save arguments number
+        parser.add_argument("--base-path", type=str, default="",
+                            help="Specify base paths for the integration test logs.")
+        parser.add_argument("--test-type", type=str, default="integ-test", help="Specify test type of this.")
+        parser.add_argument("--output-path", type=str, help="Specify the path location for the test-run manifest.")
+        parser.add_argument("--test-run-id", type=int, help="The unique execution id for the test")
+        parser.add_argument("--component", type=str, dest="components", nargs='*', help="Test a specific component or components instead of the entire distribution.")
```

Comment:
I don't think we are testing anything here. Did you mean generate repo for only 1 component rather than all components from test manifest?

Reply:
Yeah I was thinking we inherit the arguments from test workflow. So if we only tested some specific components, it's not necessary to generate the test-run manifest for all components as for the rest of the components, their data would be absent.